### PR TITLE
AddinGenerator support to use RevitVersionSelector

### PR DESCRIPTION
**Pull request by sharadkjaiswal**
- Updated the AddinGenerator utility to use DynamoRevitVersionSelector.dll as the entry point in Dynamo.addin file.
- Dynamo version selector can look at all installed Dynamo products and ensure that everything is listed on Revit as addin.
- DynamoAddinGenerator makes use of DynamoInstallDetective library to get all the Dynamo installed on the system
- It creates addin for the latest Dynamo installed on the system if the latest Dynamo doesn't support specific Revit version, then generates addin for the last/latest Dynamo install that supports this specific Revit version.
- If it is executed with /uninstall flag then it will skip generating addin for the current product for which path is provided as a parameter.

### CODE REVIEW

**Comment by sharadkjaiswal** [comment]
@aparajit-pratap PTAL

File: tools/DynamoAddinGenerator/Types.cs [hunk 1]
```diff
@@ -139,28 +118,49 @@ public DynamoAddinData(IRevitProduct product, IDynamoInstall latestDynamoInstall
...
+        /// <returns>DynamoAddinData</returns>
+        public static DynamoAddinData Create(IRevitProduct revit, DynamoProducts dynamos, string dynamoUninstallPath)
+        {
+            //Iterate in reverse order to find the first dynamo that is supported for
+            //this revit product
+            var products = dynamos.Products.Reverse();
+            foreach (var p in products)
+            {
+                //If the current product is being uninstalled, don't generate addin data
+                if (string.Equals(p.InstallLocation, dynamoUninstallPath))
```
Comment (ke-yu) [inline tools/DynamoAddinGenerator/Types.cs]: Not quite sure if it is safe to do string comparison. The following link just for your reference:
http://stackoverflow.com/questions/7344978/verifying-path-equality-with-net

**Comment by ke-yu** [comment]
Not familiar with install/uninstall process :-(. @ikeough , you may help to review the change.

**Comment by sharadkjaiswal** [comment]
Thanks @ke-yu I have updated the path comparison logic. @ikeough Please take a look when you get time. For the moment, I am going ahead and merging it so that @riteshchandawar can get started with testing. I'll also add few more unit tests.

**Pull request merged**